Fixes players and objects clipping out of stage with wall/ground portal

westonCoder opened the pull request:
- this fix was implemented by simply clamping the x and y of the localPoint to -0.1-0.1 and -0.2-0.2 respectively.
- objects can no longer clip through stage in the wall->ground portal configuration
- player can no longer clip through stage in the wall->ground portal configuration
- tested on N64 hardware, no issues.

Fixes #13
Fixes #3

westonCoder commented:
https://user-images.githubusercontent.com/71656782/227805078-886b3d0f-a889-4f6a-99ce-dd2001e3b0b8.mp4

https://user-images.githubusercontent.com/71656782/227805099-6b8f57a2-d477-44c2-aed8-95ffe87a9560.mp4

westonCoder commented:
On second thought this actually doesn't fix all clipping cases, so I think I will continue to work on it and make another push when it's more complete of a fix.

Would not recommend merging.

lambertjamesd commented:
I like this solution but I would like to see it constrain an oval shape instead of a rectangular shape.

The portal is exactly twice as tall as it is wide so you could take the local point, divide the y by 2 then constrain the point inside a circle then double the y after constraining.

westonCoder commented:
so I believe I implemented your oval change in latest merge.

I also added a plane in front of the entry and exit portal where the player will be teleported from/to. this makes it to where the player can no longer walk into the wall while standing right inside the portal. At this point I think its pretty stable to merge if you wanted to . I tested on n64 and could not get a box or myself to clip out of the stage.

westonCoder commented:
oh I also reduced the max portal speed by half because it was going faster than I think the actual game went

westonCoder commented:
Fixes #3

westonCoder commented:
yeah honestly im not happy with this implementation. I'll close and re-open when im happy with it.
